Initial edit to labeling strategy

mjbrender opened the pull request:
Fixes #1066 

Summary of changes:
- Adding a TL;DR for contributors around labels
- Describing namespaces and how we use them 
- A few typos worth editing

@intelsdi-x/snap-maintainers

mjbrender commented:
Ready for review @intelsdi-x/snap-maintainers

IzabellaRaulin commented on CONTRIBUTING.md:
```diff
@@ -63,21 +63,43 @@ When reporting an issue, details are key. Include the following:
+
+As a maintainer: 
+* An issue is opened stating that Snap is not working. Upon review, the maintainer finds it is an issue with a plugin. She will label the issue with **type/wrong-repo** and open a new issue under the plugin where she tags the original issue reporter, links the original issue and labels it **bug** (which is available in plugins repositories).
+* An issue is opened stating that Snap is not working. It turns out to be related to a Snap's functionality. The maintainer will label it **type/bug**. She takes on the bug, labels in **reviewed/work-in-progress**. She finds it maps to the Scheduler package and adds additional context with **component/scheduler**. She opens a PR while she fixes this issue. In the PR body, it says "Fixes #" for th ...
```
Probably **reviewed/work-in-progress** should be replaced by [**reviewed/in-progress** ](https://github.com/intelsdi-x/snap/labels/reviewed%2Fin-progress)

mjbrender replied:
👍

IzabellaRaulin commented on CONTRIBUTING.md:
```diff
@@ -63,21 +63,43 @@ When reporting an issue, details are key. Include the following:
+* An issue is opened stating that Snap is not working. It turns out to be related to a Snap's functionality. The maintainer will label it **type/bug**. She takes on the bug, labels in **reviewed/work-in-progress**. She finds it maps to the Scheduler package and adds additional context with **component/scheduler**. She opens a PR while she fixes this issue. In the PR body, it says "Fixes #" for th ...
+
+And as a contributor:
+* A contributor has an idea to improve Snap. He opens an issue with guidelines on how to fix it. A maintainer likes the idea, label it **type/feature-or-enhancement**. Once a maintainer or contributor begins working on the issue, it's labeled **reviewed/in-process**. A pull request is opened and labeled **wip-work-in-progress**.
```
reviewed/in-process should be replaced by reviewed/in-progress

IzabellaRaulin commented on CONTRIBUTING.md:
```diff
@@ -63,21 +63,43 @@ When reporting an issue, details are key. Include the following:
+
+And as a contributor:
+* A contributor has an idea to improve Snap. He opens an issue with guidelines on how to fix it. A maintainer likes the idea, label it **type/feature-or-enhancement**. Once a maintainer or contributor begins working on the issue, it's labeled **reviewed/in-process**. A pull request is opened and labeled **wip-work-in-progress**.
+* A contributor has an idea to improve Snap. He opens an issue with guidelines on how to fix it and the maintainer labels it **type/feature-or-enhancement**. A maintainer believes the approach requires more user input and labels it **type/rfc** to indicate it's an open discussion on implementation. Once a maintainer or contributor begins working on the issue, it's labeled **reviewed/in-process**. ...
```
As above, **reviewed/in-process** should be replaced by **reviewed/in-progress**

mjbrender replied:
👍

IzabellaRaulin commented on CONTRIBUTING.md:
```diff
@@ -63,21 +63,43 @@ When reporting an issue, details are key. Include the following:
+For the sake of clarity, here are a few scenarios you might see play out. 
+
+As a maintainer: 
+* An issue is opened stating that Snap is not working. Upon review, the maintainer finds it is an issue with a plugin. She will label the issue with **type/wrong-repo** and open a new issue under the plugin where she tags the original issue reporter, links the original issue and labels it **bug** (which is available in plugins repositories).
```
Probably **type/wrong-repo**  should be replaced by [**reviewed/wrong-repo](https://github.com/intelsdi-x/snap/issues?q=label%3Areviewed%2Fwrong-repo+is%3Aclosed)

mjbrender replied:
👍

IzabellaRaulin commented:
@mjbrender, this is a great work, especially these scenarios which clarify how to use it - thank You for that. What do You think about adding additional scenarios to section "As a maintainer" about how to use the following labels **reviewed/needs-2nd-review** and/or **effort/small**?

mjbrender commented on CONTRIBUTING.md:
```diff
@@ -63,21 +63,43 @@ When reporting an issue, details are key. Include the following:
+* A contributor has an idea to improve Snap. He opens an issue with guidelines on how to fix it. A maintainer likes the idea, label it **type/feature-or-enhancement**. Once a maintainer or contributor begins working on the issue, it's labeled **reviewed/in-process**. A pull request is opened and labeled **wip-work-in-progress**.
+* A contributor has an idea to improve Snap. He opens an issue with guidelines on how to fix it and the maintainer labels it **type/feature-or-enhancement**. A maintainer believes the approach requires more user input and labels it **type/rfc** to indicate it's an open discussion on implementation. Once a maintainer or contributor begins working on the issue, it's labeled **reviewed/in-process**. ...
+* A contributor wants to understand more about Snap and opens an issue. A maintainer sees its resolution will be an answer to the contributor, so she labels it **type/question**. The question is closed once an answer is given. If good ideas of how to improve Snap come up during that thread, she may open other issues and label them **type/** based on whether they are missing docs, improvements or  ...
+
```
👍

mjbrender commented:
@IzabellaRaulin, I added a few scenarios this morning and I rebased down to a single commit. Let men know if it needs more work (and thanks for reviewing!)

IzabellaRaulin commented on CONTRIBUTING.md:
```diff
@@ -63,21 +63,47 @@ When reporting an issue, details are key. Include the following:
+
+As a maintainer:
+* An issue is opened stating that Snap is not working. Upon review, the maintainer finds it is an issue with a plugin. She will label the issue with **reviewed/wrong-repo** and open a new issue under the plugin where she tags the original issue reporter, links the original issue and labels it **bug** (which is available in plugins repositories).
+* An issue is opened stating that Snap is not working. It turns out to be related to a Snap's functionality. The maintainer will label it **type/bug**. She takes on the bug, labels in **reviewed/in-progress**. She finds it maps to the Scheduler package and adds additional context with **component/scheduler**. She opens a PR while she fixes this issue. In the PR body, it says "Fixes #" for the pre ...
```
Consider adding sth about assigning to an issue, what's about "She assigns herself to this issue and opens a PR while she fixes this issue."? 

The another thing is using label **wip-do-not-merge** - I think this label has sense only when there is a work in progress (as the name suggests). So consider adding something about that this label should be used in special case when pull request is incomplete (for example it was made just to show a portion of works, but some things are still under work) and this label should be removed after the work is finished/code is completed.

IzabellaRaulin commented on CONTRIBUTING.md:
```diff
@@ -63,21 +63,47 @@ When reporting an issue, details are key. Include the following:
+As a maintainer:
+* An issue is opened stating that Snap is not working. Upon review, the maintainer finds it is an issue with a plugin. She will label the issue with **reviewed/wrong-repo** and open a new issue under the plugin where she tags the original issue reporter, links the original issue and labels it **bug** (which is available in plugins repositories).
+* An issue is opened stating that Snap is not working. It turns out to be related to a Snap's functionality. The maintainer will label it **type/bug**. She takes on the bug, labels in **reviewed/in-progress**. She finds it maps to the Scheduler package and adds additional context with **component/scheduler**. She opens a PR while she fixes this issue. In the PR body, it says "Fixes #" for the pre ...
+* As PR is opened that will change Snap functionality (examples at [#977](https://github.com/intelsdi-x/snap/issues/977) & [#803](https://github.com/intelsdi-x/snap/issues/803)). A maintainer labels it **reviewed/needs-2nd-review** and proceeds with the normal code review. If the initial maintainer labels LGTM, another maintainer must review it again. A discussion must take place with a technical ...
```
Probably it should be "https://github.com/intelsdi-x/snap/pull/803" instead of  "https://github.com/intelsdi-x/snap/issues/803".  Also, please verify if PR#977 is a good example - this label **reviewed/needs-2nd-review** wasn't used on this PR.

IzabellaRaulin commented on CONTRIBUTING.md:
```diff
@@ -63,21 +63,47 @@ When reporting an issue, details are key. Include the following:
+* An issue is opened stating that Snap is not working. Upon review, the maintainer finds it is an issue with a plugin. She will label the issue with **reviewed/wrong-repo** and open a new issue under the plugin where she tags the original issue reporter, links the original issue and labels it **bug** (which is available in plugins repositories).
+* An issue is opened stating that Snap is not working. It turns out to be related to a Snap's functionality. The maintainer will label it **type/bug**. She takes on the bug, labels in **reviewed/in-progress**. She finds it maps to the Scheduler package and adds additional context with **component/scheduler**. She opens a PR while she fixes this issue. In the PR body, it says "Fixes #" for the pre ...
+* As PR is opened that will change Snap functionality (examples at [#977](https://github.com/intelsdi-x/snap/issues/977) & [#803](https://github.com/intelsdi-x/snap/issues/803)). A maintainer labels it **reviewed/needs-2nd-review** and proceeds with the normal code review. If the initial maintainer labels LGTM, another maintainer must review it again. A discussion must take place with a technical ...
+* A PR is opened which changes the metadata structure for a plugin. A maintainer labels it **reviewed/needs-2nd-review** and adds whatever **breaking-change/** labels are appropriate. If the initial maintainer labels LGTM, another maintainer must review it again. A discussion must take place with a technical lead before merging. This corresponding issue is added to a milestone that corresponds wi ...
```
Consider put as an example `https://github.com/intelsdi-x/snap/pull/872` instead of `https://github.com/intelsdi-x/snap/issues/871`.

IzabellaRaulin commented on CONTRIBUTING.md:
```diff
@@ -63,21 +63,47 @@ When reporting an issue, details are key. Include the following:
+* An issue is opened stating that Snap is not working. It turns out to be related to a Snap's functionality. The maintainer will label it **type/bug**. She takes on the bug, labels in **reviewed/in-progress**. She finds it maps to the Scheduler package and adds additional context with **component/scheduler**. She opens a PR while she fixes this issue. In the PR body, it says "Fixes #" for the pre ...
+* As PR is opened that will change Snap functionality (examples at [#977](https://github.com/intelsdi-x/snap/issues/977) & [#803](https://github.com/intelsdi-x/snap/issues/803)). A maintainer labels it **reviewed/needs-2nd-review** and proceeds with the normal code review. If the initial maintainer labels LGTM, another maintainer must review it again. A discussion must take place with a technical ...
+* A PR is opened which changes the metadata structure for a plugin. A maintainer labels it **reviewed/needs-2nd-review** and adds whatever **breaking-change/** labels are appropriate. If the initial maintainer labels LGTM, another maintainer must review it again. A discussion must take place with a technical lead before merging. This corresponding issue is added to a milestone that corresponds wi ...
+* A PR is opened that edits a small amount of markdown or string output text. A maintainer labels it **effort/small**, gives it a quick review to ensure it renders, writes LGTM and merges it themselves (example: [#1139](https://github.com/intelsdi-x/snap/issues/1136)).
```
To be consistent, an example should point to a pull request - consider using some of these [PRs effor/small]( https://github.com/intelsdi-x/snap/pulls?q=is%3Apr+is%3Aclosed+label%3Aeffort%2Fsmall)

IzabellaRaulin commented on CONTRIBUTING.md:
```diff
@@ -63,21 +63,47 @@ When reporting an issue, details are key. Include the following:
+And as a contributor:
+* A contributor has an idea to improve Snap. He opens an issue with guidelines on how to fix it. A maintainer likes the idea, label it **type/feature-or-enhancement**. Once a maintainer or contributor begins working on the issue, it's labeled **reviewed/in-progress**. A pull request is opened and labeled **wip-do-not-merge**.
+* A contributor has an idea to improve Snap. He opens an issue with guidelines on how to fix it and the maintainer labels it **type/feature-or-enhancement**. A maintainer believes the approach requires more user input and labels it **type/rfc** to indicate it's an open discussion on implementation. Once a maintainer or contributor begins working on the issue, it's labeled **reviewed/in-progress** ...
+* A contributor wants to understand more about Snap and opens an issue. A maintainer sees its resolution will be an answer to the contributor, so she labels it **type/question**. The question is closed once an answer is given. If good ideas of how to improve Snap come up during that thread, she may open other issues and label them **type/** based on whether they are missing docs, improvements or  ...
```
For this scenario the issue https://github.com/intelsdi-x/snap/issues/964 could be a good example - but it's only proposition.

IzabellaRaulin commented:
@mjbrender, could you explain me why **wip-do-not-merge** is always used while creating a PR in these exemplary scenarios? I thought it was dedicated to special cases, when we would like to show some part of code before it's ready, so open PR is not completed and some things are still under work. An example when whole effort was splitted and made step by step was https://github.com/intelsdi-x/snap/pull/872 (see description of this PR which presents phases of work). Please, correct me if I am wrong and also skip my comments above. 

Thank You a lot for that additional scenarios. Amazing work!

mjbrender commented:
@IzabellaRaulin the importance of *wip-do-not-merge* is process oriented. We should label here when we open a PR but have no intention to have it reviewed for merge. Opening a PR early will help us communicate across teams on what's in flight and filtering out this label gives maintainers an ability to filter for merge-able PRs.

IzabellaRaulin commented on README.md:
```diff
@@ -292,28 +292,7 @@ Amongst the many awesome contributors, there are the maintainers. These maintain
 * Committing to reviewing pull requests, issues, and addressing comments/questions as quickly as possible
 * Acting as a point of contact for questions
 
-Just tag **@intelsdi-x/snap-maintainers** if you need to get some attention on an issue. If at any time, you don't get a quick enough response, reach out to any of the following team members directly:
```
@mjbrender, could You explain me why the table with snap-maintainers has been removed?

mjbrender replied:
@IzabellaRaulin I was asked by @lynxbat to remove it since it consistently gets stale. We favor of pointing people to Slack. I'd be game to reintroduce it when we generate the table automatically from members of the `@intelsdi-x/snap-maintainers` team.

IzabellaRaulin replied:
Ok, that sounds great.

IzabellaRaulin commented:
LGTM

mjbrender commented:
Thanks @IzabellaRaulin 👍
